cql3/statements/list_permissions_statement.cc

```diff
@@ -106,15 +105,15 @@ cql3::statements::list_permissions_statement::execute(
 
     const auto& as = *state.get_client_state().get_auth_service();
 
-    return do_with(make_resource_filter(), [this, &as](const auto& resource_filter) {
+    return do_with(make_resource_filter(), [this, &as, metadata = std::move(metadata)](const auto& resource_filter) {
         return auth::list_filtered_permissions(
                 as,
                 _permissions,
                 _role_name,
-                resource_filter).then([](std::vector<auth::permission_details> all_details) {
+                resource_filter).then([metadata = std::move(metadata)](std::vector<auth::permission_details> all_details) {
```

Comment:
This will be a copy, not a move, because the surrounding lambda is not `mutable` (should be probably made mutable)

Reply:
> TBH this entire function could also greatly benefit from coroutinization

I agree but that's true for 99% of auth code I need to stop somewhere (for now at least :))

> Although I'm not sure how result_set works

We use this constructor here:
```
result_set::result_set(std::vector<lw_shared_ptr<column_specification>> metadata_)
    : _metadata(::make_shared<metadata>(std::move(metadata_)))
{ }
```

so I think it will keep metadata alive.


> because the surrounding lambda is not mutable (should be probably made mutable)

will make it mutable